opcua/server/internal_server.py

```diff
@@ -310,18 +318,45 @@ def close_session(self, delete_subs=True):
 
     def activate_session(self, params):
         self.logger.info("activate session")
+
         result = ua.ActivateSessionResult()
         if self.state != SessionState.Created:
             raise utils.ServiceError(ua.StatusCodes.BadSessionIdInvalid)
+
+        # FIXME: Get the server available UserIdentityToken policy's ids. I don't how to get it in other way.
+        # iserver_policy_ids = ["anonymous", "certificate_basic256sha256", "username"]
+        # Note that it is different from server._policyIDs = ["Anonymous", "Basic256Sha256", "Username"]
+        iserver_policy_ids = []
+        edp = self.iserver.get_endpoints()[0]  # FIXME: is it the same list whatever endpoint ?
+        for token in edp.UserIdentityTokens:
+            iserver_policy_ids.append(token.PolicyId)
+
+        # Check if params UserIdentityToken is authorized by the server.
+        params_id_token = params.UserIdentityToken
+        if params_id_token.PolicyId not in iserver_policy_ids:
+            raise utils.ServiceError(ua.StatusCodes.BadIdentityTokenInvalid)
+
+        # Route to the correct manager
+        if isinstance(params_id_token, ua.AnonymousIdentityToken):
+            pass  # TODO: Call AnonymousIdentityToken Manager
```

Comment:
we should at least check that "anonymous" is in the list of Policies and return an UaError if not

Reply:
Can't we use the user_manager for that? maybe just have a bool member called something like "allow_anonymous"? not sure I have not looked at that code for very long...